relayer-cli/src/bin/hermes/main.rs

```diff
@@ -8,9 +8,15 @@ use ibc_relayer_cli::components::enable_ansi;
 
 /// Boot Cli
 fn main() -> eyre::Result<()> {
-    if enable_ansi() {
+    if compact_logs() {
+        oneline_eyre::install()?;
+    } else if enable_ansi() {
         color_eyre::install()?;
     }
 
     abscissa_core::boot(&APPLICATION);
 }
+
+fn compact_logs() -> bool {
+    matches!(std::env::var("RUST_BACKTRACE").as_deref(), Ok("" | "0"))
```

Comment:
I think in this case if `RUST_BACKTRACE` is not set, the error case `Err(_)` will return false when it should be true.

This should be clearer with comment explaining what should happen in different branches:

```rust
fn main() -> eyre::Result<()> {
    if enable_ansi() {
        // If we are in a terminal supporting color, display
        // full error logs in color
        color_eyre::install()?;
    } else if !backtrace_enabled() {
        // Else if we are piping logs to file and backtrace is *not* enabled,
        // display errors in single line.
        oneline_eyre::install()?;
    } else {
        // Else backtrace is enabled and we are piping to logs, so use the
        // default error report handler, which displays multiline errors
        // without color.
    }

    abscissa_core::boot(&APPLICATION);
}

fn backtrace_enabled() -> bool {
    match std::env::var("RUST_BACKTRACE").as_deref() {
        Ok("" | "0") | Err(_) => false,
        Ok(_) => true,
    }
}
```

Reply:
Good catch, thanks! Shouldn't we invert the first two branches? ie. if backtraces are not enabled, we use one-line logs, otherwise if there is a tty we use colors, else we use the default reporter

Reply:
I think it depends on what we want the default behavior to be. Most of the time the relayer will run without `RUST_BACKTRACE` set, even for terminal cases. Do we want those to display one-line errors as well?

Reply:
Most of the time the relayer will be piping to logs. So one line logs should be the default case.